### ISSUE TITLE
[TODO]: Clean hardcoded stuff

### ISSUE THREAD

**Issue by retraigo** [retraigo/la-classy#2]
- [x] Keep gradient descent as a separate function instead of hardcoding.
- [x] Keep loss functions as separate functions instead of hardcoding.
- [ ] Allow both f32 and f64 inputs.
